src/TesApi.Web/BatchScheduler.cs

```diff
@@ -440,11 +518,23 @@ private async Task AddBatchJobAsync(TesTask tesTask)
                 tesTask.SetFailureReason("BatchClientException", string.Join(",", exception.Data.Values), exception.Message, exception.StackTrace);
                 logger.LogError(exception, $"TES task: {tesTask.Id} BatchClientException.Message: {exception.Message} {string.Join(",", exception?.Data?.Values)}");
             }
-            catch (BatchException exception) when (exception.InnerException is Microsoft.Azure.Batch.Protocol.Models.BatchErrorException batcnErrorException && @"ActiveJobAndScheduleQuotaReached".Equals(batcnErrorException.Body.Code, StringComparison.OrdinalIgnoreCase))
+            catch (BatchException exception) when (exception.InnerException is Microsoft.Azure.Batch.Protocol.Models.BatchErrorException batcnErrorException && IsJobQuotaException(batcnErrorException.Body.Code))
+            {
+                tesTask.SetWarning(batcnErrorException.Body.Message.Value, Array.Empty<string>());
+                logger.LogDebug($"Not enough quota available for task Id {tesTask.Id}. Reason: {batcnErrorException.Body.Message.Value}. Task will remain in queue.");
```

Comment:
should we make this Information so that someone is aware the reason is due to quota?

Reply:
Yes, although we are currently filtering out Information as well.